Assets/ConsentManagementProvider/Plugins/iOS/Source/SwiftBridge.swift

```diff
@@ -177,33 +162,17 @@ import UIKit
         myVendorAccepted = .Unknown
     }
     
-    @objc public func onGDPRPrivacyManagerTap() {
-        if config.gdprPmId != nil {
-            (consentManager != nil) ? 
-                consentManager?.loadGDPRPrivacyManager(withId: config.gdprPmId!) :
-                self.runCallback(callback: self.callbackOnErrorCallback, arg: "Library was not initialized correctly!")
-        } else {
-            self.runCallback(callback: self.callbackOnErrorCallback, arg: "Tried to load GDPR pm without gdpr pm id")
-        }
-    }
-    
-    @objc public func onCCPAPrivacyManagerTap() {
-        if config.ccpaPmId != nil {
-            (consentManager != nil) ? 
-                consentManager?.loadCCPAPrivacyManager(withId: config.ccpaPmId!) :
-                self.runCallback(callback: self.callbackOnErrorCallback, arg: "Library was not initialized correctly!")
-        } else {
-            self.runCallback(callback: self.callbackOnErrorCallback, arg: "Tried to load CCPA pm without ccpa pm id")
-        }
-    }
-    
-    @objc public func onUSNATPrivacyManagerTap() {
-        if config.usnatPmId != nil {
-            (consentManager != nil) ?
-                consentManager?.loadUSNatPrivacyManager(withId: config.usnatPmId!) :
-                self.runCallback(callback: self.callbackOnErrorCallback, arg: "Library was not initialized correctly!")
+    @objc public func loadPrivacyManager(campaignType: Int, pmId: String, tab: SPPrivacyManagerTab) {
+        if (consentManager != nil) {
+            switch CAMPAIGN_TYPE(rawValue: campaignType) {
+            case .GDPR: consentManager!.loadGDPRPrivacyManager(withId: pmId, tab: tab)
+            case .CCPA: consentManager!.loadCCPAPrivacyManager(withId: pmId, tab: tab)
+            case .USNAT: consentManager!.loadUSNatPrivacyManager(withId: pmId, tab: tab)
```

Comment:
were you can use:
```swift
guard let consentManager = consentManager else {
  self.callbacks.RunCallback(callbackType: .OnErrorCallback, arg: "Library was not initialized correctly!")
  return
}
// from here on we don't need to force unwrap consentManager(!) since the guard let took care of it
```

Reply:
Also, could we think about a way of making `consentManager` not optional? This way we don't have to keep checking if it's nil on every method.

Reply:
@andresilveirah having `guard` is a good idea, thanks! 
As for making `consentManager` not optional – for now I cannot imagine how to do it, because to make it not optional we have to initialize it with values, and we cannot declare and initialize them in one place since we wait for the initialization-related data to come from Unity side of a bridge.
I have to think a little bit about it.